### PR TITLE
Choose random test port and properly format response

### PR DESCRIPTION
**Pull request by krisalyssa**

### CODE REVIEW

File: test/nectar_test.exs
Comment (jeffutter) [inline test/nectar_test.exs]: Ahh, that's a clearer way to show that.

Comment (jeffutter) [inline test/nectar_test.exs:6]: Ahh, that's a clearer way to show that.

1.) The wikipedia page said.

> Many Linux kernels use the port range 32768 to 61000

That you can't access > 61000 via `/proc` I'm not sure if that is a problem on some systems? That's why I went with a lower range.

2.) I think it can take a bit of time for it to fail to open the port when it is in use. I'm not sure if it is a good idea to try all of them (it might block for a _long_ time). Also, you would probably have better luck picking random ones, in case something is using a range of ports. Maybe?

File: lib/nectar/worker.ex
Comment (jeffutter) [inline lib/nectar/worker.ex:167]: This takes a lot of lines. Maybe looking it up from a map would be just as good? And we could put it in the `request.ex` file?

I don't think it should hold up the PR but this file is getting big.

**Review by jeffutter — CHANGES_REQUESTED** [review]
👍 finding the line ending thing. The only thing on the Pr is maybe the port finding in the test.

File: test/nectar_test.exs
Comment (krisalyssa) [reply test/nectar_test.exs:6]: 1. I missed the lower suggested limit of 61000. That's an easy fix.

2. It should only take time if a significant fraction of the ports in the ephemeral range are in use. My concern with the random check that we had before was the arbitrary limit (I think it was 5?) on the number of retries. There's a small but non-zero chance that, regardless of how many times you pick a port randomly, you're going to pick one that's in use when there are lots of unused ones. Maybe something like this?

```elixir
49152..61000 |> Enum.shuffle |> Enum.find(fn port -> start_server(port) end)
```

File: lib/nectar/worker.ex
Comment (krisalyssa) [reply lib/nectar/worker.ex:167]: That can probably be condensed down to a macro, TBH. If it's going to move anywhere, it should be to `response.ex` instead of `request.ex`.

File: test/nectar_test.exs
Comment (jeffutter) [reply test/nectar_test.exs:6]: Yeah, that will work. There is still a chance that it could pick wrong many times and hang for a while but it's pretty unlikely.

File: lib/nectar/worker.ex
Comment (jeffutter) [reply lib/nectar/worker.ex:167]: Heh, yeah I meant `response.ex` not awake yet I guess :)

**Comment by jeffutter** [comment]
Approving pending the port selection changes.

**Review by jeffutter — APPROVED** [review]

**Pull request merged**
